offchain/http-server/src/lib.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright Cartesi Pte. Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+mod config;
+pub use config::HttpServerConfig;
+
+// Re-exporting prometheus' Registry.
+pub use prometheus_client::registry::Registry;
+
+// Re-exporting prometheus metrics.
+// Add any other metrics to re-export here.
+pub use prometheus_client::metrics::counter::Counter;
+// End of metrics to re-export.
+
+use axum::{routing::get, Router};
+use prometheus_client::encoding::text::encode;
+use std::{
+    net::SocketAddr,
+    sync::{Arc, Mutex},
+};
+
+/// If neither the healthcheck and the metrics are enabled, there is no
+/// point in starting the server.
+pub fn should_start(config: &HttpServerConfig) -> bool {
+    config.healthcheck_enabled || config.metrics_enabled
+}
+
+/// Starts a HTTP server with two endpoints: /healthz and /metrics.
+///
+/// The `Registry` parameter is a `prometheus` type used for metric tracking.
+pub async fn start(
+    config: HttpServerConfig,
+    registry: Registry,
```

Comment:
The metrics will be collected whether the server is up or not.
So we send the registry regardless and let the start function `if` it out.

Otherwise we would have to put an `if` before calling the function to "optionalize" it, and then have another if to check whether the server should start with that endpoint.